### PR TITLE
feat: rate limit for otp generation

### PR DESCRIPTION
**Pull request by tshuli**
## Problem

Closes [#129](https://github.com/datagovsg/formsg-private/issues/129)

## Tests
- [ ] Create form with verified mobile field and test that verification SMSes still work

### CODE REVIEW

File: src/app/routes/api/v3/forms/public-forms.verification.routes.ts [hunk 1]
```diff
@@ -58,4 +60,7 @@ PublicFormsVerificationRouter.route(
  */
 PublicFormsVerificationRouter.route(
   '/:formId([a-fA-F0-9]{24})/fieldverifications/:transactionId([a-fA-F0-9]{24})/fields/:fieldId([a-fA-F0-9]{24})/otp/generate',
-).post(VerificationController.handleGenerateOtp)
+).post(
+  limitRate({ max: rateLimitConfig.sendAuthOtp }),
```
Comment (mantariksh) [inline src/app/routes/api/v3/forms/public-forms.verification.routes.ts:64]: I think we need to apply the same limit on the old endpoint `/:transactionId([a-fA-F0-9]{24})/otp`

Comment (tshuli) [reply src/app/routes/api/v3/forms/public-forms.verification.routes.ts:64]: thanks! i added it

File: docs/DEPLOYMENT_SETUP.md [hunk 2]
```diff
@@ -240,7 +240,7 @@ The app applies per-minute, per-IP rate limits at specific API endpoints as a se
 | Variable | Description |
 | :-------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `SUBMISSIONS_RATE_LIMIT` | Per-minute, per-IP request limit for each submissions endpoint. The limit is applied separately for the email mode and encrypt mode endpoints. |
-| `SEND_AUTH_OTP_RATE_LIMIT` | Per-minute, per-IP request limit for the endpoint which requests for new login OTPs for the admin console. |
+| `SEND_AUTH_OTP_RATE_LIMIT` | Per-minute, per-IP request limit for the endpoint which requests for new login OTPs for the admin console or mobile number verifications. |
```
Comment (mantariksh) [inline docs/DEPLOYMENT_SETUP.md:243]: the same limit will apply to email verifications too!

File: src/app/config/schema.ts [hunk 3]
```diff
@@ -284,7 +284,7 @@ export const optionalVarsSchema: Schema<IOptionalVarsSchema> = {
       env: 'SUBMISSIONS_RATE_LIMIT',
     },
     sendAuthOtp: {
-      doc: 'Per-minute, per-IP request limit for OTPs to log in to the admin console',
+      doc: 'Per-minute, per-IP request limit for OTPs to log in to the admin console or mobile number verifications',
```
Comment (mantariksh) [inline src/app/config/schema.ts:287]: email verification as well

**Review by mantariksh — APPROVED** [review]

**Pull request merged**